### PR TITLE
Adds Redis configuration to docs

### PR DESCRIPTION
**Pull request by stevsmit**
Issue: https://issues.redhat.com/browse/PROJQUAY-3888

### CODE REVIEW

File: modules/config-fields-redis.adoc [hunk 1]
```diff
@@ -35,20 +33,18 @@
...
 **Example:** +
 `quay-server.example.com` 
-|{nbsp}{nbsp}{nbsp}**.port**  + 
-{nbsp}{nbsp}{nbsp}(Required)| Number | The port at which Redis is accessible + 
-  {nbsp} + 
-**Example:** +
-`6379`
-|{nbsp}{nbsp}{nbsp}**.password** | String | The port at which Redis is accessible + 
-  {nbsp} + 
+|**.password** | String | The port at which Redis is accessible +
```
Comment (syed) [inline modules/config-fields-redis.adoc:40]: password is the password for redis when using encrypted mode

File: modules/config-fields-redis.adoc [hunk 2]
```diff
@@ -10,20 +10,18 @@
...
```
Comment (syed) [inline modules/config-fields-redis.adoc:17]: same here, fix the description for password

File: modules/config-fields-redis.adoc [hunk 3]
```diff
@@ -59,9 +55,11 @@ BUILDLOGS_REDIS:
     host: quay-server.example.com
     password: strongpassword
     port: 6379
+    ssl: false
```
Comment (syed) [inline modules/config-fields-redis.adoc:58]: `ssl: false` and `password: somepassword` cannot exist together. You can have:
```
ssl: false
```
or 
```
ssl: true
password: somepassword
```

**Comment by syed** [comment]
There are 3 possibilities for modelcache
1. memcache (this is the default) you don't need to configure anything
2. single redis instance. Config for this:
```
    DATA_MODEL_CACHE_CONFIG:
      engine: rediscluster
      redis_config:
        host: <host>
        port: <port>
        password: <password if ssl is true>
        ssl: <true | false >
```
3. clustered redis instance. Config for this is:
```
    DATA_MODEL_CACHE_CONFIG:
      engine: rediscluster
      redis_config:
        startup_nodes:
          - host: <cluster-host>
            port: <port>
        password: <password if ssl: true>
        read_from_replicas: <true|false>
        skip_full_coverage_check: <true | false>
        ssl: <true | false > 
```

File: modules/config-fields-redis.adoc [hunk 4]
```diff
@@ -34,34 +32,40 @@
...
     password: strongpassword
     port: 6379
-```
\ No newline at end of file
+    ssl: true
+----
+
+[NOTE]
+====
+If `ssl` is set to `false`, you will not need a password for Redis.
```
Comment (LiZhang19817) [inline modules/config-fields-redis.adoc:70]: can't agree on this point, customer can config redis authentication with password even not enabled SS

Comment (stevsmit) [reply modules/config-fields-redis.adoc]: Fixed

File: modules/config-fields-redis.adoc [hunk 5]
```diff
@@ -34,34 +32,35 @@
...
 BUILDLOGS_REDIS:
     host: quay-server.example.com
     password: strongpassword
     port: 6379
+    ssl: true
 
 USER_EVENTS_REDIS:
     host: quay-server.example.com
     password: strongpassword
     port: 6379
```
Comment (LiZhang19817) [inline modules/config-fields-redis.adoc:64]: when enable ssl, the default port number is 6380, here is an example of azure redis service connection string "quay37.redis.cache.windows.net:6380,password=k2NJbsnpRqsWBdb4fhIUbYuNhi3LDHtyuAzCaNV2Zqs=,ssl=True,abortConnect=False"

File: modules/config-fields-redis.adoc [hunk 6]
```diff
@@ -34,34 +32,35 @@
...
 
 == Example redis configuration
 
-```
+[source,yaml]
+----
 BUILDLOGS_REDIS:
     host: quay-server.example.com
     password: strongpassword
     port: 6379
```
Comment (LiZhang19817) [inline modules/config-fields-redis.adoc:58]: when enable ssl, the default port number is 6380,

**Pull request merged**